doc/specs/#11000 - Marks/Shell-Integration-Marks.md

```diff
@@ -0,0 +1,474 @@
+marks manually. These can be thought of like "bookmarks" - a user indicated
+region that means something to the user.
+
+* [ ] `addMark`: add a mark to the buffer. If there's a selection, place the
+  mark covering at the selection. Otherwise, place the mark on the cursor row.
+  - [x] `color`: a color for the scrollbar mark. (in [#12948])
+  - [ ] `category`: one of `{"prompt", "error", "warning", "success", "info"}`
+  - [ ] `dismissSelection`: `bool`, default `true`. If true, dismiss the
+    selection (if there is one) after marking it. If false, leave the text
+    selected.
+    - This is considered for something like pattern matching, where a user might
+      want to match on "error: ", mark it with an error mark, and then also
+      colorize it. 
+* [ ] `scrollToMark`
+  - [x] `direction`: `["first", "previous", "next", "last"]`  (in [#12948])
+  - [ ] [#13455] - `highlight`: `bool`, default false. Display a temporary
+    highlight around the mark when scrolling to it. ("Highlight" != "select")
+    - If the mark has prompt/command/output sections, only select the prompt and command.
+    - If the mark has zero width (i.e. the user just wanted to bookmark a line),
+      then highlight the entire row.
+  - [ ] [#13449] - `center` or some other setting that controls how the mark is scrolled in.
+    - Maybe `top` (current) /`center` (as proposed) /`nearestEdge` (when
+      scrolling down, put the mark at the bottom line of viewport , up -> top
+      line)?
+  - [ ] `category`: `flags({categories}...)`, default `"all"`. Only scroll to
+    one of the categories specified (e.g. only scroll to the previous error,
+    only the previous prompt, or just any mark)
+* [x] `clearMark`: Remove any marks in the selected region (or at the cursor
+  position)  (in [#12948])
+* [x] `clearAllMarks`: Remove all the marks from the buffer.  (in [#12948])
+
+
+#### Selecting commands & output
+
+_Inspired by a long weekend of manually copying .csv output from the Terminal to
+a spreadsheet, only to discover that we rejected [#4588] some years ago._
+
+* [x] `selectCommand(direction=[prev, next])`: Starting at the active selection
+  anchor, (or the cursor if there's no selection), select the command that
+  starts before/after this point (exclusive).  Probably shouldn't wrap around
+  the buffer.
+  * Since this will create a selection including the start of the command,
+    performing this again will select the _next_ command (in whatever
+    direction).
+* [x] `selectOutput(direction=[prev, next])`: same as above, but with command outputs.
+
+A convenient workflow might be a `multipleActions([selectOutput(prev),
+copy()])`, to quickly select the previous commands output.
+
+### Per-profile settings
+
+* [x] `autoMarkPrompts`: `bool`, default `false`.  (in [#12948])
+* [ ] `showFindMatchesOnScrollbar`: `bool`, default `true`.
+* [ ] `showMarksOnScrollbar`: `bool` or `flags({categories}...)` 

```

Comment:
oh I _love_ this flags thing

Reply:
i _LOVE_ the flags idea. I want to get msbuild to emit errors as marked lines (or, until then, #15700 )